### PR TITLE
Improve Source Service

### PR DESCRIPTION
**Pull request by tomschr**
I've restructured this chapter and try to improve the structure. Here are the changes:

* Introduce an "About Source Service"
* Give an example with callouts
* Add xml:id's
* Convert sect2 into variablelist entries
* Change order of some sect1s

I'm not completely satisfied with the result yet, but it's a first start.

### CODE REVIEW

File: xml/obs_source_service.xml [hunk 1]
```diff
@@ -7,154 +7,299 @@
...
+  <para>
+   Source Services may be used to validate sources. This can be definied
+   as:
+  </para>
+  <itemizedlist>
+   <listitem>
+    <formalpara>
+     <title>Per Package</title>
+     <para>
+      Useful when the packager wants to validate the downloaded sources
```
Comment (sknorr) [inline xml/obs_source_service.xml]: ...validate **whether** the downloaded...

File: xml/obs_source_service.xml [hunk 2]
```diff
@@ -7,154 +7,299 @@
...
+      are really from the original maintainer.
+     </para>
+    </formalpara>
+   </listitem>
+   <listitem>
+    <formalpara>
+     <title>Per Project</title>
+     <para>
+      Useful for applying general policies which cannot get skipped in
```
Comment (sknorr) [inline xml/obs_source_service.xml]: general->project-wide (?)
get->be
"in" is definitely wrong, but i am not 100% sure of my proposal either: in->for

File: xml/obs_source_service.xml [hunk 3]
```diff
@@ -7,154 +7,299 @@
...
+      any package.
+     </para>
+    </formalpara>
+   </listitem>
+  </itemizedlist>
+  <remark>toms 2017-08-31: Hmn, not sure how to rephrase the following
+  sentence...</remark>
```
Comment (sknorr) [inline xml/obs_source_service.xml]: issues i see:
* the weird dichotomy of "validating files" versus "redoing a certain action," 
* the wording makes validation sound like an accidental thing ("can happen")
* redo->repeat

This is an attempt to make sense of this but content-wise it is possibly/likely wrong:
"You can validate sources using either of two methods:
* By comparing checksums and metadata of the files in your repository with checksums and metadata as recorded by the maintainer.
* Alternatively, you can download the sources from a trusted location again and verify that they did not change."

however, (based off my interpretation) the use cases seem to be different for those two methods: the first guards against corruptions that happen on the way between maintainer and you (i.e. rogue ISP). the second guards against corruptions between the source project and you (i.e. rogue maintainer). interestingly, neither guards against both.

File: xml/obs_source_service.xml [hunk 4]
```diff
@@ -7,154 +7,299 @@
...
+   <listitem>
+    <para>
+     Compares the downloaded file (<filename>krabber-1.0.tar.gz</filename>)
+     against the SHA256 checksum.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     If enabled, upgrade the source to a newer version available online.
+     This service is usually not executed unless when you run
```
Comment (sknorr) [inline xml/obs_source_service.xml]: remove "usually" (as the only condition when it is run is named in the second half of the sentence)
remove "when"

File: xml/obs_source_service.xml [hunk 5]
```diff
@@ -7,154 +7,299 @@
...
+
+ <sect1 xml:id="sec.obs.sserv.about">
+  <title>About Source Service</title>
+  <para>
+   Source Services are tools to validate, generate or modify sources in a
+   trustable way. They are designed as smallest possible tools and can be
+   combined following the powerful idea of the classic UNIX design.
+  </para>
+  <para>
+   Design goals of source services were:
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Right, try to rephrase them in a more user-facing way (although the changes are minor, maybe you can have a final look on this).

File: xml/obs_source_service.xml [hunk 6]
```diff
@@ -7,154 +7,299 @@
...
+   </listitem>
+  </itemizedlist>
+  <para>
+   For using source services you need (refer to <xref
+    linkend="ex.obs.sserv.struct"/>):
+  </para>
+  <itemizedlist>
+   <listitem>
+    <para>
+     An XML file named <filename>_services</filename>.
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed

File: xml/obs_source_service.xml [hunk 7]
```diff
@@ -7,154 +7,299 @@
...
+   </listitem>
+   <listitem>
+    <para>
+     A <tag>service</tag> element which uses the specific services with
+     optional parameters.
+    </para>
+   </listitem>
+  </itemizedlist>
+  <example xml:id="ex.obs.sserv.struct">
+   <title>General Structure of a <filename>_service</filename> File</title>
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed

File: xml/obs_source_service.xml [hunk 8]
```diff
@@ -7,154 +7,299 @@
...
+    </callout>
+    <callout arearefs="co.obs.sserv.struct.name">
+     <para>
+      The service name. The service is a script that is stored in the
+      <filename>/usr/lib/obs/service</filename> directory.
+     </para>
+    </callout>
+    <callout arearefs="co.obs.sserv.struct.mode">
+     <para>
+      A mode of service, see <xref linkend="sec.obs.sserv.mode"/>.
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Hmn, not sure if this is applicable here.

File: xml/obs_source_service.xml [hunk 9]
```diff
@@ -7,154 +7,299 @@
...
+     </para>
+    </callout>
+    <callout arearefs="co.obs.sserv.struct.param">
+     <para>
+      One or more parameter which are passed to the script definied in
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed

File: xml/obs_source_service.xml [hunk 10]
```diff
@@ -7,154 +7,299 @@
...
+  <para>
+   The example above will execute the script:
+  </para>
+  <screen><command>/usr/lib/obs/service/<replaceable
+   >MY_SCRIPT</replaceable></command> --<replaceable>PARAMETER1</replaceable> <replaceable>PARAMETER1_VALUE</replaceable> --outdir <replaceable>DIR</replaceable></screen>
  </sect1>
- <sect1>
+
+ <sect1 xml:id="sec.obs.sserv.mode">
   <title>Different Modes of Services</title>
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed. Or should it better be named "Service Modes"?

File: xml/obs_source_service.xml [hunk 11]
```diff
@@ -7,154 +7,299 @@
...
   <para> Each service can be used in a special mode defining when it should run
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed.

File: xml/obs_source_service.xml [hunk 12]
```diff
@@ -7,154 +7,299 @@
...
+      and skip the generated files. In case they differ, they get generated
+      and added on the server.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><literal>localonly</literal> Mode</term>
+    <listitem>
+     <para>
+      This mode is running the service locally. The result gets committed
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed.

File: xml/obs_source_service.xml [hunk 13]
```diff
@@ -7,154 +7,299 @@
...
+      as standard files and not named  with _service: prefix.
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed, used `literal`.

File: xml/obs_source_service.xml [hunk 14]
```diff
@@ -7,154 +7,299 @@
...
+       os="opensuse;novell"> (as is always the case at <link
+        xlink:href="https://build.opensuse.org"/>)</phrase>.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><literal>disabled</literal> Mode</term>
+    <listitem>
+     <para>
+      The disabled mode is neither running the service locally or on the
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed

File: xml/obs_source_service.xml [hunk 15]
```diff
@@ -7,154 +7,299 @@
...
-   order. Each service may define a list of parameters and a mode. The project
-   wide services get called after the per package defined services. The
-   _service file is an XML file like this example:</para>
-   <screen>
-&lt;services&gt;
+
+ <sect1 xml:id="sec.obs.sserv.validate">
+  <title>Defining Services for Validation</title>
+  <para>
+   Source Services may be used to validate sources. This can be definied
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Both fixed.

File: xml/obs_source_service.xml [hunk 16]
```diff
@@ -7,154 +7,299 @@
...
+   as:
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: FIxed.

File: xml/obs_source_service.xml (hunk 1 above)
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed.

File: xml/obs_source_service.xml [hunk 17]
```diff
@@ -7,154 +7,299 @@
...
+   one during build. </para>
+ </sect1>
+
+ <sect1 xml:id="sec.obs.sserv.create">
+  <title>Creating Source Service Definitions</title>
+  <para>
+   Source services are definied in the <filename>_service</filename> file
+   and are either part of the package sources or used project-wide.
+   <remark>toms 2017-08-31: Not sure where this _project belongs; where
+   is this _project package?
```
[the remark this answers is not in the document]
Comment (adrianschroeter) [reply xml/obs_source_service.xml]: that link is a reproducer of a webui/api bug.

A _project package object exists always in the backend, but we don't have a database object usually (minus this bug). The webui is not showing it therefore, but you can check it out via osc, when you specify it explicit: "osc co $project _project"

File: xml/obs_source_service.xml (hunk 2 above)
Comment (tomschr) [reply xml/obs_source_service.xml]: Yes, used project-wide. Fixed.

File: xml/obs_source_service.xml [hunk 18]
```diff
@@ -7,154 +7,299 @@
...
+     service run must never harm the system of a user. </para>
+   </listitem>
+   <listitem>
+    <para> services shall be designed in a way to avoid unnecessary commits.
+     This means there shall be no time-dependent changes. In case the package
+     already contains the same file, the newly generated file must be dropped.
+    </para>
+   </listitem>
+   <listitem>
+    <para> local services can be added and used by everybody. </para>
```
Comment (adrianschroeter) [inline xml/obs_source_service.xml]: add also the "buildtime" services here: Services running local or inside the build environment can get created, added and used by everbody.

File: xml/obs_source_service.xml [hunk 19]
```diff
@@ -7,154 +7,299 @@
...
+   </listitem>
+   <listitem>
+    <para> server side services must be installed by the admin of the OBS
+     server. </para>
```
Comment (adrianschroeter) [inline xml/obs_source_service.xml]: These are the only ones which run on the server side and may have network access.

Maybe we should phrase it also like this to make clear that this depends on the mode and not the service itself:
"Services running in default or server side mode must be ...."

File: xml/obs_source_service.xml (hunk 3 above)
Comment (tomschr) [reply xml/obs_source_service.xml]: Ok, I've used your change and added the last paragraph as a remark. We can revise it in the next round.

File: xml/obs_source_service.xml [hunk 20]
```diff
@@ -7,154 +7,299 @@
...
+   state "broken". Or validation can happen by redoing a certain action and
+   store the result as new file as <command>download_files</command> is doing.
+   In this case the newly generated file will be used instead of the committed
+   one during build. </para>
+ </sect1>
+
+ <sect1 xml:id="sec.obs.sserv.create">
+  <title>Creating Source Service Definitions</title>
+  <para>
+   Source services are definied in the <filename>_service</filename> file
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed

File: xml/obs_source_service.xml [hunk 21]
```diff
@@ -7,154 +7,299 @@
...
+   </listitem>
+   <listitem>
+    <para> local services can be added and used by everybody. </para>
+   </listitem>
+   <listitem>
+    <para> server side services must be installed by the admin of the OBS
+     server. </para>
+   </listitem>
+   <listitem>
+    <para> service can be defined per package or project wide. </para>
```
Comment (adrianschroeter) [inline xml/obs_source_service.xml]: The use of a service can be defined ...

(not the definition what the service is)

File: xml/obs_source_service.xml [hunk 22]
```diff
@@ -7,154 +7,299 @@
...
+   Source services are definied in the <filename>_service</filename> file
+   and are either part of the package sources or used project-wide.
+   <remark>toms 2017-08-31: Not sure where this _project belongs; where
+   is this _project package?
+   "It is either part of the package sources
+   or used project-wide when stored inside the <filename>_project</filename>
+   package." </remark>
+  </para>
+  <para> The <filename>_service</filename> file contains a list of services
+   which get called in the listed order. Each service may define a list of
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed

File: xml/obs_source_service.xml [hunk 23]
```diff
@@ -7,154 +7,299 @@
...
+   </listitem>
+   <listitem>
+    <para>
+     Compares the downloaded file (<filename>krabber-1.0.tar.gz</filename>)
+     against the SHA256 checksum.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     If enabled, upgrade the source to a newer version available online.
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed.

File: xml/obs_source_service.xml (hunk 11 above)
[the remark this answers is not in the document]
Comment (adrianschroeter) [reply xml/obs_source_service.xml]: I noticed that I complained mostly at my own text ;)

File: xml/obs_source_service.xml (hunk 4 above)
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed with the last change from https://github.com/openSUSE/obs-docu/pull/81#discussion_r136389296

File: xml/obs_source_service.xml [hunk 24]
```diff
@@ -7,154 +7,299 @@
...
+     This service is usually not executed unless when you run
+     <command>osc service disabledrun</command>.
+    </para>
+   </listitem>
+  </orderedlist>
  </sect1>
+
  <sect1>
   <title>Dropping a Source Service Again</title>
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed.

File: xml/obs_source_service.xml (hunk 18 above)
Comment (tomschr) [reply xml/obs_source_service.xml]: Ok, used your 2nd sentence.

File: xml/obs_source_service.xml (hunk 19 above)
Comment (tomschr) [reply xml/obs_source_service.xml]: Ok, added.

File: xml/obs_source_service.xml (hunk 21 above)
Comment (tomschr) [reply xml/obs_source_service.xml]: Ok, done

File: xml/obs_source_service.xml [hunk 25]
```diff
@@ -7,154 +7,299 @@
...
-  <sect2>
-   <title><literal>disabled</literal> Mode</title>
-   <para> The disabled mode is neither running the service locally or on the
-    server side. It can be used to temporarily disable the service but keeping
-    the definition as part of the service definition. Or it can be used to
-    define the way how to generate the sources and doing so by manually calling
-    <command>osc service disabledrun</command> The result will
-    get committed as standard files again. </para>
-  </sect2>
+  <variablelist>
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: I have to try it out. 😉

[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Ok, I've tried it and it doesn't really work. Although I like this idea, the table raises more questions as it solves. For the time being, I will integrate the table as a comment, so we can enable it in the future.

File: xml/obs_source_service.xml [hunk 26]
```diff
@@ -7,154 +7,299 @@
...
+   package." </remark>
+  </para>
+  <para> The <filename>_service</filename> file contains a list of services
+   which get called in the listed order. Each service may define a list of
+   parameters and a mode.
+   The project wide services get called after the per package defined
+   services.
+   </para>
+  <para>
+   The file is in XML format like this example:
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_source_service.xml]: Ok, I've rephrased it and added:

```xml
The <filename>_service</filename> file is in XML format and looks like this:
```

File: xml/obs_source_service.xml [hunk 27]
```diff
@@ -145,6 +145,80 @@
...
+       </para>
+      </footnote></entry>
+      <entry>Before each build<xref linkend="fn.obs.sserv.mode.buildtime"/></entry>
+      <entry></entry>
+     </row>
+     <row>
+      <entry><literal>disabled</literal></entry>
+      <entry>Never</entry>
+      <entry>Never</entry>
+      <entry><remark>toms 2017-09-01: </remark></entry>
```
Comment (sknorr) [inline xml/obs_source_service.xml]: empty remark. remove?

File: xml/obs_source_service.xml [hunk 28]
```diff
@@ -286,6 +360,7 @@
    "It is either part of the package sources
    or used project-wide when stored inside the <filename>_project</filename>
    package." </remark>
+   For project-wide services,
```
Comment (sknorr) [inline xml/obs_source_service.xml]: Dangling sentence beginning

File: xml/obs_source_service.xml (hunk 25 above)
[the remark this answers is not in the document]
Comment (sknorr) [reply xml/obs_source_service.xml]: Yeah, makes sense. But having the commented table in there, so we can enable it later if it proves helpful is a good idea.

File: xml/obs_source_service.xml [hunk 29]
```diff
@@ -18,46 +18,65 @@
...
   <para>
-   Design goals of source services were:
+   Source services allow:
   </para>
   <itemizedlist>
    <listitem>
-    <para> server side generated files must be easy to identify and must not be
+    <para>
+     Server side generated files are easy to identify and are not
      modifiable by the user. This way others user can trust them to be
```
Comment (sknorr) [inline xml/obs_source_service.xml]: others->other

File: xml/obs_source_service.xml [hunk 30]
```diff
@@ -207,46 +226,60 @@
...
+   at different levels:
   </para>
   <itemizedlist>
    <listitem>
     <formalpara>
      <title>Per Package</title>
      <para>
-      Useful when the packager wants to validate the downloaded sources
-      are really from the original maintainer.
+      Useful when the packager wants to validate wheather the downloaded
```
Comment (sknorr) [inline xml/obs_source_service.xml]: wheather->whether

File: xml/obs_source_service.xml [hunk 31]
```diff
@@ -207,46 +226,60 @@
...
   <para>
-   Validation can happen by validating files, for example, using the
-    <systemitem class="service">verify_file</systemitem> or <systemitem
-      class="service">source_validator</systemitem> service.
-   These services just fail in the error case which leads to the build
-   state "broken". Or validation can happen by redoing a certain action and
-   store the result as new file as <command>download_files</command> is doing.
-   In this case the newly generated file will be used instead of the committed
-   one during build. </para>
+   You can validate sources using either of two methods:
```
Comment (sknorr) [inline xml/obs_source_service.xml]: i am a bit queasy about whether my interpretation of the previous content is correct here -- could you add a fixme remark or so?

File: xml/obs_source_service.xml (hunk 27 above)
Comment (tomschr) [reply xml/obs_source_service.xml]: Yes, the table is commented anyway so that doesn't hurt.

File: xml/obs_source_service.xml (hunk 29 above)
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed.

File: xml/obs_source_service.xml (hunk 30 above)
Comment (tomschr) [reply xml/obs_source_service.xml]: Fixed

File: xml/obs_source_service.xml (hunk 31 above)
Comment (tomschr) [reply xml/obs_source_service.xml]: I've added a remark already in one of the last commits, see the next lines.

File: xml/obs_source_service.xml (hunk 28 above)
Comment (tomschr) [reply xml/obs_source_service.xml]: Argh, yes. I missed that part.

**Comment by tomschr** [comment]
Merged with 0136e8d20c5 and f350d3402

**Pull request closed without merging**